### PR TITLE
[CINN] Fix test_dot_merger segmentation fault

### PR DESCRIPTION
**Pull request by FisherWY**
### PR types
Others

### PR changes
Others

### Description
Pcard-72511

`test_dot_merger`和`test_dot_merger_pass`在CI和本地环境中会随机出现`segmentation fault`错误，导致单测随机挂，经排查后发现是`ginac`库相关的问题，现避免使用`MathIsZero`绕过

image

### CODE REVIEW

File: paddle/cinn/ir/tensor.cc [hunk 1]
```diff
@@ -553,14 +553,14 @@ ir::Tensor _Tensor_::Reshape(const std::vector<Expr> &shape,
   auto selft = Tensor(const_cast<ir::_Tensor_ *>(this));
 
   {
-    Expr this_num_elements = Expr(1);
-    for (auto &e : this->shape) this_num_elements = this_num_elements * e;
+    int32_t this_num_elements = 1;
+    for (auto &e : this->shape)
```
Comment (zhhsplendid) [inline paddle/cinn/ir/tensor.cc]: Try to add braces for `for` loops even if they only contain one statement.

Reference:

https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching

Comment (FisherWY) [reply paddle/cinn/ir/tensor.cc]: Done✅

**Review by 6clc — APPROVED** [review]
LGTM

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**